share/eds/rsmt2d.go

```diff
@@ -21,8 +21,11 @@ type Rsmt2D struct {
 }
 
 // Size returns the size of the Extended Data Square.
-func (eds *Rsmt2D) Size(context.Context) int {
-	return int(eds.Width())
+func (eds *Rsmt2D) Size(context.Context) (int, error) {
+	if eds.ExtendedDataSquare == nil {
+		return 0, fmt.Errorf("extended data square is not initialized")
+	}
```

Comment:
Not necessary and can be removed. Other methods don't check for nil eds.ExtendedDataSquare.